src/clr-angular/utils/component/is-attribute-set.ts

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+export function isAttributeSet(value: string | boolean | number | object): boolean {
+  // for null just return false no need to check anything
+  if (value === null) {
+    return false;
+  }
+  if (typeof value === 'string') {
+    // Empty string is valid, 'true' as string is also valid
+    return value === '' || value === 'true';
+  }
+  if (typeof value === 'number') {
+    // Must not be 0
+    return value >= 1;
+  }
+  if (typeof value === 'object') {
+    return Object.keys(value).length > 0 && value.constructor === Object;
+  }
+  // Boolean value will be read as it is, everything else is false
```

Comment:
This is actually backwards from HTML attribute semantics. Any value that is provided is technically `true`. Even `inverse="false"` is technically true. We can just do a basic check on truthyness of `value` with the one except of an empty string which evaluates to `false`.

I think we should adhere to technical semantics and this basically becomes the following.

```
if (value === '' || value) { 
  // true 
} else { 
  // false 
};
```

Reply:
I'm addressing this in my later comment so will close that one.